api/php/DBgateways/RestaurantDbGateway.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+include_once "php/config.php";
+
+class RestaurantDbGateway {
+
+    // private $servername = "localhost";
+    // private $username = "root";
+    // private $password = "foody";
+    // private $db = "foody";
+
+
+    private $dbConnection = null;
+
+    public function __construct() {
+        $this->dbConnection = new DbConnecter();
+    }
+
+    // public function getRestaurants () {
+
+    // }
+
+    public function validateRestaurant (string $restaurantName, string $restaurantAddress, string $restaurantPhone, string $restaurantWebsite) {
+        $restaurantName = $this->dbConnection->cleanInput($restaurantName);
+        $restaurantAddress = $this->dbConnection->cleanInput($restaurantAddress);
+        $restaurantPhone = $this->dbConnection->cleanInput($restaurantPhone);
+        $restaurantWebsite = $this->dbConnection->cleanInput($restaurantWebsite);
+        $restaurantName = $this->dbConnection->cleanInput($restaurantName);
+        $restaurantAddress = $this->dbConnection->cleanInput($restaurantAddress);
+
+        $sql = "CALL foody.sp_validateRestaurant ('$restaurantName', '$restaurantAddress', '$restaurantPhone', '$restaurantWebsite');";
+
+        $result = $this->dbConnection->returnQuery($sql);
+
+        $output = (int)mysqli_fetch_assoc($result)["restaurantId"];
+
+
+        $this->dbConnection->cleanSPResults();
+
+        return $output;
+    }
+
+    public function saveRestaurantReview (int $restaurantId, int $userId, string $review, int $deliciousnessScore, int $serviceScore, int $experienceScore, int $pricingScore, float $pricingValue) {
+        $review = $this->dbConnection->cleanInput($review);
+
+        $sql = "CALL foody.sp_saveReview ($restaurantId, $userId, '$review', $deliciousnessScore, $serviceScore, $experienceScore, $pricingScore, $pricingValue);";
+
+        $result = $this->dbConnection->returnQuery($sql);
+        $output = mysqli_fetch_assoc($result)["result"];
+
+        $this->dbConnection->cleanSPResults();
+
+        return $output;
+    }
+
+    public function addRestaurantImage (int $restaurantId, int $reviewId, string $imageName, string $imageSize, string $imageType, string $image) {
+        $sql = "INSERT INTO foody.restaurantImages (restaurantId, reviewId, imageName, imageSize, imageType, imageEncoded)
+            VALUES ('$restaurantId', '$reviewId', '$imageName', '$imageSize', '$imageType', '$image')";
```

Comment:
SQL query on a single line makes code base hard to read. We should write our queries more with this type of formatting style: 
INSERT INTO foody.restaurant_images ( 
  restaurant_id, 
  review_id, 
  image_name, 
  image_size, 
  image_type, 
  image_encoded
) VALUES (
  '$restaurantId', 
  '$reviewId', '
   $imageName', 
  '$imageSize',  
  '$imageType',  
  '$image' )";

I also said we use camel case for programming languages. However, I failed to mention that MySQL has its own coding styling/convention. Typically, it is in all lowercase with underscores to represent spaces. This makes it more readable. https://anandarajpandey.com/2015/05/10/mysql-naming-coding-conventions-tips-on-mysql-database/